python/cudf/cudf/core/buffer/exposure_tracked_buffer.py

```diff
@@ -139,103 +41,50 @@ def mark_exposed(self) -> None:
-            An object implementing the CUDA Array Interface.
-        exposed : bool, optional
-            Mark the buffer as permanently exposed.
-
-        Returns
-        -------
-        ExposureTrackedBuffer
-            Buffer representing the same device memory as `data`
-        """
-        ret = super()._from_device_memory(data)
+    def _from_device_memory(cls, data: Any, exposed: bool) -> Self:
+        ret = super()._from_device_memory(data, exposed=exposed)
         ret._exposed = exposed
         ret._slices = weakref.WeakSet()
         return ret
 
-    def _getitem(self, offset: int, size: int) -> BufferSlice:
-        return BufferSlice(base=self, offset=offset, size=size)
-
     @property
     def __cuda_array_interface__(self) -> Mapping:
         self.mark_exposed()
         return super().__cuda_array_interface__
 
-    def __repr__(self) -> str:
-        return (
-            f"<ExposureTrackedBuffer exposed={self.exposed} "
-            f"size={format_bytes(self._size)} "
-            f"ptr={hex(self._ptr)} owner={repr(self._owner)}>"
-        )
-
 
-class BufferSlice(ExposureTrackedBuffer):
-    """A slice (aka. a view) of a exposure tracked buffer.
+class ExposureTrackedBuffer(Buffer):
+    """An exposure tracked buffer.
 
     Parameters
     ----------
-    base
-        The exposure tracked buffer this slice refers to.
+    owner
+        The owning exposure tracked buffer this refers to.
     offset
-        The offset relative to the start memory of base (in bytes).
+        The offset relative to the start memory of owner (in bytes).
     size
         The size of the slice (in bytes)
-    passthrough_attributes
-        Name of attributes that are passed through to the base as-is.
     """
 
+    _owner: ExposureTrackedBufferOwner
+
     def __init__(
         self,
-        base: ExposureTrackedBuffer,
-        offset: int,
-        size: int,
-        *,
-        passthrough_attributes: Container[str] = ("exposed",),
+        owner: ExposureTrackedBufferOwner,
+        offset: int = 0,
+        size: Optional[int] = None,
     ) -> None:
-        if size < 0:
-            raise ValueError("size cannot be negative")
-        if offset < 0:
-            raise ValueError("offset cannot be negative")
-        if offset + size > base.size:
-            raise ValueError(
-                "offset+size cannot be greater than the size of base"
-            )
-        self._base = base
-        self._offset = offset
-        self._size = size
-        self._owner = base
-        self._passthrough_attributes = passthrough_attributes
-        base._slices.add(self)
-
-    def __getattr__(self, name):
-        if name in self._passthrough_attributes:
-            return getattr(self._base, name)
-        raise AttributeError(
-            f"{self.__class__.__name__} object has no attribute {name}"
-        )
+        super().__init__(owner=owner, offset=offset, size=size)
+        self._owner._slices.add(self)
```

Comment:
So this asserts that `owner` is an `ExposureTrackedBuffer`.

Reply:
Yes, this is enforced by `as_buffer()`